src/server/Story2sketch.js

```diff
@@ -254,22 +259,55 @@ export default class Story2sketch {
   positionSymbols() {
     let xOffset = 0;
 
-    for (const { id } of this.sortedViewports) {
-      let yOffset = 0;
+    for (const kind of Object.keys(this.symbolsByViewport)) {
+      for (const { id } of this.sortedViewports) {
+        let yOffset = 0;
+
+        const symbols = this.symbolsByViewport[kind][id];
+
+        for (const [index, symbol] of symbols.entries()) {
+          // Skip failed symbols
+          if (symbol) {
+            symbol.frame.x = xOffset;
+            symbol.frame.y = yOffset;
+            this.sketchPage.layers.push(symbol);
+            yOffset += this.tallestByStory[index] + this.symbolGutter;
+          }
+        }
+
+        xOffset += this.widestByViewport[id] + this.symbolGutter;
+      }
+    }
+  }
```

Comment:
The changes here make this function incompatible with the original layout. Since I'm keen to retain the original layout, we could:

1. Make this functional, so it accepts an object called `symbolsByViewport` (which is actually keyed on viewport - see other comment regarding adding `symbolsByKind` variable), and returns an array of symbols.

```js
positionSymbols(symbolsByViewport) {
    let layers = [];
    let xOffset = 0;

    for (const { id } of this.sortedViewports) {
      let yOffset = 0;

      const symbols = symbolsByViewport[id];

      for (const [index, symbol] of symbols.entries()) {
        // Skip failed symbols
        if (symbol) {
          symbol.frame.x = xOffset;
          symbol.frame.y = yOffset;
          layers.push(symbol);
          yOffset += this.tallestByStory[index] + this.symbolGutter;
        }
      }

      xOffset += this.widestByViewport[id] + this.symbolGutter;
    }
    
    return layers;
}

```

2. Add an additional method called `positionSymbolsByKind`, which loops over `symbolsByKind` and calls `positionSymbols` for each one, like

```js
positionSymbolsByKind() {
  for (const kind of Object.keys(this.symbolsByViewport)) {
    const symbolsByViewport = this.symbolsByKind[kind];

    this.sketchPage.layers = [
      ...this.sketchPage.layers,
      ...this.positionSymbols(symbolsByViewport)
    ];
  }
}
```
3. Add an additional method called `positionSymbolsByViewport`:

```js
positionSymbolsByViewport() {
  this.sketchPage.layers = [
    ...this.sketchPage.layers,
    ...this.positionSymbols(this.symbolsByViewport)
  ];
}
```

Reply:
Integrated this code ✅